diff --git a/node_modules/@luma.gl/core/dist/adapter/canvas-context.js b/node_modules/@luma.gl/core/dist/adapter/canvas-context.js
index d4436e1..9d39cbf 100644
--- a/node_modules/@luma.gl/core/dist/adapter/canvas-context.js
+++ b/node_modules/@luma.gl/core/dist/adapter/canvas-context.js
@@ -166,7 +166,8 @@ export class CanvasContext {
     }
     /** Returns the biggest allowed framebuffer size. @todo Allow the application to limit this? */
     getMaxDrawingBufferSize() {
-        const maxTextureDimension = this.device.limits.maxTextureDimension2D;
+        const maxTextureDimension = this.device?.limits?.maxTextureDimension2D;
+        if (!maxTextureDimension) return [4096, 4096];
         return [maxTextureDimension, maxTextureDimension];
     }
     /** Update the canvas drawing buffer size. Called automatically if props.autoResize is true. */
